src/instructlab/model/evaluate.py

```diff
@@ -213,29 +220,32 @@ def qa_pairs_to_qna_to_avg_scores(qa_pairs: list[dict]) -> dict:
     return qna_to_avg_scores
 
 
-def launch_server(
-    ctx,
-    model,
-    model_name,
-    max_workers,
-) -> tuple:
+def launch_server(ctx, model, model_name, max_workers, backend, max_ctx_size) -> tuple:
     # pylint: disable=import-outside-toplevel
     # First Party
-    from instructlab.model.backends import backends
-
-    if not ctx.obj.config.serve.backend:
-        ctx.obj.config.serve.backend = backends.VLLM
-    if ctx.obj.config.serve.backend == backends.VLLM:
-        ctx.obj.config.serve.vllm.vllm_args.extend(["--served-model-name", model_name])
-    elif ctx.obj.config.serve.backend == backends.LLAMA_CPP:
-        # mt_bench requires a larger context size
-        ctx.obj.config.serve.llama_cpp.max_ctx_size = 5120
+
+    # we shouldn't be in the business of editing a user's cfg.
+    # with eval, this is tricky since there are a few things we want to help them out with if they do not exist.
+    # if the user did not set a backend, or model_path, set that for them. Everything else needs to be in the cfg. or in the cmd as flag overrides.
+    temp_serve = copy.copy(ctx.obj.config.serve)
+    if backend is None:
+        temp_serve.backend = backends.VLLM
+    if backend == backends.VLLM:
+        temp_serve.vllm.vllm_args.extend(["--served-model-name", model_name])
+        temp_serve.vllm.max_ctx_size = max_ctx_size
+    if backend == backends.LLAMA_CPP:
+        if max_ctx_size != 5120:
+            click.secho(
+                f"Max ctx size of {max_ctx_size} does not match the requirement for the serving backend. For vllm the ctx_size should be 4096 and for llama-cpp it should be 5120. Please edit your config or use the --max-ctx-size flag.",
```

Comment:
Ditto what Dan said. I think we can remove this warning altogether.

Reply:
I am confused on this one. Previously we were setting the ctx FOR the user to 5120. IF we remove this error, then we let them continue with the wrong ctx size? is that a breaking change?

Reply:
The context size needs to be 5120 for everything.  Since it's set in config for llama, we need to override.  For vllm, we don't set a max so we inherit the max of the model.

Reply:
ah I see. hm. I think the best thing to do here is default to the cfg and flag then and remove the error, yeah. I wonder if we should just switch the ctx default to 5120. Would that hurt anyone else?

Reply:
> Would that hurt anyone else?

Unfortunately it would.  If the model is configured to a lower context size max, vllm will error on api requests with a larger context.